hw/ip/otbn/dv/otbnsim/sim/state.py

```diff
@@ -66,9 +73,41 @@ def set_next_pc(self, next_pc: int) -> None:
         assert(self.is_pc_valid(next_pc))
         self._pc_next_override = next_pc
 
-    def set_rnd_data(self, rnd_data: int) -> None:
+    def step_edn(self, rnd_data: int) -> None:
+        # Take the new data
         self._new_rnd_data = rnd_data
 
+        # Collect 32b packages in a 256b variable
+        self.rnd_256b = ((self.rnd_256b << 32) + self._new_rnd_data)
+        self.rnd_256b = self.rnd_256b & ((1 << 256) - 1)
+        self._new_rnd_data = None
```

Comment:
I think we can get rid of the `_new_rnd_data` field. Also, it would be nice to have a range check on the `rnd_data` argument. Maybe something like this:
```suggestion
    def step_edn(self, rnd_data: int) -> None:
        assert 0 <= rnd_data < (1 << 32)
        # Take 32 bits of new data, collecting it into a 256 bit accumulator
        self.rnd_256b = ((self.rnd_256b << 32) | rnd_data) & ((1 << 256) - 1)
```

Alternatively, we could do this differently and collect up rnd_data into an array of 32 bit values. Then we could use the length of the array as `rnd_256b_counter` as well. No strong feelings, but maybe it's worth thinking about?